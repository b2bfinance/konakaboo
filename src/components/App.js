import React, { Component } from 'react';
import { connect } from 'react-redux';
import Row from './Row';
import Col from './Col';
import ProductList from './Product';
import Filter from './Filter';
import { loadProducts } from '../actions/products';

export const StyledAppContainer = Row.extend`
  font-family: ${props => props.theme.mainFontFamily};
  font-weight: ${props => props.theme.mainNormalFontWeight};
  font-size: ${props => props.theme.mainFontSize};
  color: ${props => props.theme.mainColor};

  * {
    box-sizing: border-box;
  }
`;

export class AppContainer extends Component {
  componentDidMount() {
    const { dispatch } = this.props;
    dispatch(loadProducts());
  }

  render() {
    const { filterCount, products, hasFetched, isFetching, error } = this.props;

    return (
      <StyledAppContainer>
        {filterCount > 0 && <Filter />}
        <Row>
          <Col phone="100">
            <ProductList
              products={products}
              hasFetched={hasFetched}
              isFetching={isFetching}
              error={error}
            />
          </Col>
        </Row>
      </StyledAppContainer>
    );
  }
}

export const mapStateToProps = state => {
  const { filters, products } = state;

  return {
    filterCount: filters.available.length,
    isFetching: products.isFetching,
    hasFetched: !!products.items,
    products: products.items,
    error: products.error
  };
};

export default connect(mapStateToProps)(AppContainer);
