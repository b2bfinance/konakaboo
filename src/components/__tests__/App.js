import React from 'react';
import renderer from 'react-test-renderer';
import { render, shallow } from 'enzyme';
import { StyledAppContainer, AppContainer, mapStateToProps } from '../App';

test('StyledAppContainer renders correctly', () => {
  expect(
    render(<StyledAppContainer theme={stubData.theme} />)
  ).toMatchSnapshot();
});

test('AppContainer with filters renders correctly', () => {
  expect(
    shallow(<AppContainer dispatch={jest.fn()} filterCount={1} />)
  ).toMatchSnapshot();
});

test('AppContainer without filters renders correctly', () => {
  expect(
    shallow(<AppContainer dispatch={jest.fn()} filterCount={0} />)
  ).toMatchSnapshot();
});

test('Should dispatch loadProducts', () => {
  const mockDispatch = jest.fn();
  shallow(<AppContainer dispatch={mockDispatch} />);
  expect(mockDispatch.mock.calls.length).toBe(1);
});

test('mapStateToProps maps the state correctly', () => {
  const stubState = {
    filters: stubData.filters.withSingleAndMultiChoiceChosen,
    products: {
      error: false,
      isFetching: false,
      items: stubData.products.response.data
    }
  };

  expect(mapStateToProps(stubState)).toEqual({
    filterCount: stubState.filters.available.length,
    isFetching: false,
    hasFetched: true,
    products: stubState.products.items,
    error: false
  });
});
