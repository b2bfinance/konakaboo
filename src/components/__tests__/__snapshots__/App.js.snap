// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppContainer renders correctly with filters 1`] = `
<Styled(styled.div)>
  <List />
  <styled.div>
    <styled.div
      phone="100"
    >
      <Component />
    </styled.div>
  </styled.div>
</Styled(styled.div)>
`;

exports[`AppContainer renders correctly without filters 1`] = `
<Styled(styled.div)>
  <styled.div>
    <styled.div
      phone="100"
    >
      <Component />
    </styled.div>
  </styled.div>
</Styled(styled.div)>
`;

exports[`Wrapper renders correctly 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
}

.c0 {
  font-family: Roboto,sans-serif;
  font-weight: 300;
  font-size: 1rem;
  color: #37474f;
}

.c0 * {
  box-sizing: border-box;
}

<div
  class="c0 c1"
>
  
</div>
`;
